components/util/time.rs

```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::time::Duration;
+use std::{u32, u64};
+
+pub fn duration_from_seconds(secs: f64) -> Duration {
+    const NANOS_PER_SEC: f64 = 1_000_000_000.0;
```

Comment:
We could just pull the constant out, so that it'll be useful for future methods.